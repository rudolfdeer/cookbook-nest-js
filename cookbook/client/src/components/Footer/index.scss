@import '../../styles/index.scss';

.footer {
  background-color: $main;

  a {
    color: $white;
  }

  .wrapper {
    padding: 10px;
  }

  &__nav {
    &__list {
      @include nav-list;

      &__item {
        margin-right: 72px;
        color: $white;

        a {
          color: $white;
        }

        &:hover {
          a {
            color: $primary-yellow;
          }
        }

        &:last-child {
          margin-right: 0;
        }
      }
    }
  }

  &__container {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
    min-height: 97px;
    font-size: 16px;
    background: $main;
  }

  &__logo {
    width: 112px;
    height: 33px;
    background: url('../../assets/images/logo-black.png') center no-repeat;
  }

  &__email {
    a {
      color: $primary-yellow;
    }
  }

  &__study {
    display: flex;
    flex-wrap: nowrap;

    &__project {
      color: $white-dirty;
      margin-right: 18px;
    }
  }

  &__lang {
    &__btn {
      background: none;
      color: $white;
      border: none;

      &:first-child {
        border-right: 1px solid $white-dirty;
      }

      &:hover {
        cursor: pointer;
      }
    }
  }
}

@media (max-width: 1199px) {
  .footer {
    &__container {
      flex-direction: column;
      align-items: center;

      a {
        margin-bottom: 24px;
      }
    }

    &__email {
      margin-bottom: 24px;
    }

    &__nav {
      width: 100%;
      margin-bottom: 24px;

      &__list {
        justify-content: center;
    
      }
    }

    &__study {
      margin-bottom: 24px;

     
    }

    &__lang {
      &__btn {
        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 0;
        }
      }
    }
  }
}

