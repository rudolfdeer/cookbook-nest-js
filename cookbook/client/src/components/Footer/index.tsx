import * as React from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import ROUTES from '../../constants/routes';

import './index.scss';

export default function Footer(): JSX.Element {
  const { t, i18n } = useTranslation();

  const changeLanguage = (e: React.MouseEvent) => {
    const target = e.target as HTMLButtonElement;
    i18n.changeLanguage(target.value);
    localStorage.setItem('lang', target.value);
  };

  return (
    <footer className="footer">
      <div className="wrapper">
        <div className="footer__container">
          <Link to={ROUTES.HOME}>
            <div className="footer__logo" />
          </Link>
          <nav className="footer__nav">
            <ul className="footer__nav__list">
              <li className="footer__nav__list__item">
                <Link to={ROUTES.COOKBOOKS}>{t('COOKBOOKS')}</Link>
              </li>
              <li className="footer__nav__list__item">
                <Link to={ROUTES.RECIPES}>{t('RECIPES')}</Link>
              </li>
              <li className="footer__nav__list__item">
                <Link to="/">{t('ABOUT_US')}</Link>
              </li>
            </ul>
          </nav>
          <div className="footer__email">
            <a href={`mailto:${t('MY_EMAIL')}`}>{t('MY_EMAIL')}</a>
          </div>
          <div className="footer__study">
            <div className="footer__study__project">{t('PROJECT')}</div>
            <div className="footer__study__itechart">
              <a href={t('ITECHART_SITE')}>{t('ITECHART')}</a>
            </div>
          </div>
          <div className="footer__lang">
            <button
              className="footer__lang__btn"
              value="en"
              onClick={(e) => changeLanguage(e)}
            >
              English
            </button>
            <button
              className="footer__lang__btn"
              value="fr"
              onClick={(e) => changeLanguage(e)}
            >
              Fran√ßais
            </button>
          </div>
        </div>
      </div>
    </footer>
  );
}
