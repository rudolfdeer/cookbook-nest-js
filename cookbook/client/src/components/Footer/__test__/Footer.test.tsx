import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../index';

describe('footer component', () => {
  const component = shallow(<Footer />);

  it('renders the <Footer /> component', () => {
    expect(component).toMatchSnapshot();
  });
});
