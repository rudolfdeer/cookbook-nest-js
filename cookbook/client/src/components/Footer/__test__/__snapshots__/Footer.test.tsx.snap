// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer component renders the <Footer /> component 1`] = `ShallowWrapper {}`;
